# Chile Censo 2017 - Data analysis 

<p><img src="http://www.censo2017.cl/wp-content/uploads/2016/01/logo-2.png"></p>
<br>

### The Dataset:
https://redatam-ine.ine.cl/redbin/RpWebEngine.exe/Portal?BASE=CENSO_2017&lang=esp

In [2]:
import pandas as pd
import numpy as np
import matplotlib
import matplotlib.pyplot as plt
%matplotlib inline

In [3]:
ls

Censo2017_16R_ManzanaEntidad.ipynb     Censo2017_Manzanas.csv
Censo2017_Identificación_Geográfica/   LEAME.docx


In [4]:
# Load dataset
data = './Censo2017_Manzanas.csv'
manzanas = pd.read_csv(data, sep=';')
manzanas.shape # 180_499, 59)

(180499, 59)

In [18]:
# manzanas.to_excel('manzanas2.xlsx')

In [5]:
manzanas.head()

REGION  PROVINCIA  COMUNA  DC  AREA  ZC_LOC  MZ_ENT  ID_ZONA_LOC  \
0       1         11    1101   1     1       1       1         7849   
1       1         11    1101   1     1       1      10         7849   
2       1         11    1101   1     1       1      11         7849   
3       1         11    1101   1     1       1      12         7849   
4       1         11    1101   1     1       1      13         7849   

      ID_MANZENT  PERSONAS       ...       MATREC MATIRREC P05_1 P05_2 P05_3  \
0  1101011001001        15       ...            0        0     0     0     0   
1  1101011001010        70       ...            1        0    15     0     0   
2  1101011001011        36       ...            5        0    15     0     0   
3  1101011001012        65       ...            9        4    24     0     0   
4  1101011001013        39       ...            0        0     9     0     0   

  P05_4 REGION_15R PROVINCIA_15R  COMUNA_15R  ID_MANZENT_15R  
0     0          1            11        1101   1101011001001  
1     0          1            11        1101   1101011001010  
2     0          1            11        1101   1101011001011  
3     0          1            11        1101   1101011001012  
4     0          1            11        1101   1101011001013  

[5 rows x 59 columns]

In [6]:
manzanas.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 180499 entries, 0 to 180498
Data columns (total 59 columns):
REGION            180499 non-null int64
PROVINCIA         180499 non-null int64
COMUNA            180499 non-null int64
DC                180499 non-null int64
AREA              180499 non-null int64
ZC_LOC            180499 non-null int64
MZ_ENT            180499 non-null int64
ID_ZONA_LOC       180499 non-null int64
ID_MANZENT        180499 non-null int64
PERSONAS          180499 non-null int64
HOMBRES           180499 non-null object
MUJERES           180499 non-null object
EDAD_0A5          180499 non-null object
EDAD_6A14         180499 non-null object
EDAD_15A64        180499 non-null object
EDAD_65YMAS       180499 non-null object
INMIGRANTES       180499 non-null object
PUEBLO            180499 non-null object
VIV_PART          180499 non-null int64
VIV_COL           180499 non-null int64
VPOMP             180499 non-null int64
TOTAL_VIV         180499 non-null int64
C

In [7]:
manzanas.PERSONAS.describe()

count    180499.000000
mean         97.363437
std         149.647870
min           0.000000
25%          35.000000
50%          63.000000
75%         105.000000
max        9197.000000
Name: PERSONAS, dtype: float64

In [8]:
len(manzanas.HOMBRES[manzanas.HOMBRES == '*'])

3413

In [9]:
len(manzanas.MUJERES[manzanas.MUJERES == '*'])

3413

Hay 3413 Hombres y mujeres que no són identificados. Se puede considerar la proporción chilena: 1.043 Mujer x 1.0 hombre o dejar 50% (#personas/2)<br>
Las formulas abajo, calculan el numero de Hombres e mujeres desconocidos utilizadose de los 50%

In [10]:
# *********** NOTA: REACER LAS FORMULAS CON .LOC ***************
manzanas['HOMBRES'] = manzanas.apply(lambda row: round((row['PERSONAS'] / 2), 0) if row['HOMBRES'] in ('*') else row['HOMBRES'], axis=1)
manzanas['MUJERES'] = manzanas.apply(lambda row: round((row['PERSONAS'] - row['HOMBRES']), 0) if row['MUJERES'] in ('*') else row['MUJERES'], axis=1)

In [11]:
manzanas['HOMBRES'] = manzanas['HOMBRES'].astype(int)
manzanas['MUJERES'] = manzanas['MUJERES'].astype(int)

In [12]:
POP_CHILE = manzanas.PERSONAS.sum()

In [13]:
POP_HOMBRES_CHILE = manzanas.HOMBRES.sum()

In [14]:
POP_MUJERES_CHILE = manzanas.MUJERES.sum()

In [15]:
VIVENDAS_CHILE = manzanas.TOTAL_VIV.sum()

In [16]:
HOGARES_CHILE = manzanas.CANT_HOG.sum()

In [17]:
manzanas.INMIGRANTES.describe()

count     180499
unique       400
top            0
freq       95642
Name: INMIGRANTES, dtype: object

In [18]:
len(manzanas)

180499

In [19]:
len(manzanas.INMIGRANTES[manzanas.INMIGRANTES == '*'])

49220

In [20]:
len(manzanas.INMIGRANTES[manzanas.INMIGRANTES == '0'])

95642

In [21]:
len(manzanas.INMIGRANTES[manzanas.INMIGRANTES != '0'])

84857

Hay 49220 manzanas sin informacion si las personas son imigrantes. Acá se las consideró cero!

In [22]:
manzanas['INMIGRANTES'] = manzanas.apply(lambda row: 0 if row['INMIGRANTES'] in ('*') else row['INMIGRANTES'], axis=1)
manzanas['INMIGRANTES'] = manzanas['INMIGRANTES'].astype(int)

In [23]:
POP_INMIGRANTES_CHILE = manzanas.INMIGRANTES.sum()

In [24]:
print("Numero de personas en Chile:", POP_CHILE)
print("Numero de hombres en Chile:", POP_HOMBRES_CHILE)
print("Numero de mujeres en Chile:", POP_MUJERES_CHILE)
print("Numero de inmigrantes en Chile:", POP_INMIGRANTES_CHILE)
print("Numero de vivendas en Chile:", VIVENDAS_CHILE)
print("Numero de hogares en Chile:", HOGARES_CHILE)
print("Numero medio de personas por hogar en Chile:", round(POP_CHILE/HOGARES_CHILE, 2))
print("Porcentaje de inmigrantes en Chile {}%".format(round((POP_INMIGRANTES_CHILE/POP_CHILE)*100, 2)))

Numero de personas en Chile: 17574003
Numero de hombres en Chile: 8596972
Numero de mujeres en Chile: 8977031
Numero de inmigrantes en Chile: 664120
Numero de vivendas en Chile: 6499355
Numero de hogares en Chile: 5651637
Numero medio de personas por hogar en Chile: 3.11
Porcentaje de inmigrantes en Chile 3.78%
